viewer/packages/cad-styling/src/prioritized/mergeboxes/SmartMergeBoxes.ts

```diff
@@ -29,21 +28,36 @@ export class SmartMergeBoxes implements BoxClusterer {
     this.resultBoxes.pop();
   }
 
-  private shouldMergeBoxesAtIndices(i: number, j: number): boolean {
-    const BOX_MERGE_MIN_IOU = 0.15;
-    const overlap = intersectionOverUnion(this.resultBoxes[i], this.resultBoxes[j]);
-    return overlap >= BOX_MERGE_MIN_IOU;
+  private surfaceArea(box: Box3) {
+    const size = box.getSize(new Vector3());
+
+    return 2 * (size.x * size.y + size.y * size.z + size.z * size.x);
+  }
+
+  private shouldMergeBoxes(box0: Box3, box1: Box3): boolean {
+    const MAX_SURFACE_INCREASE_RATIO = 1.0;
+    const MAX_SURFACE_INCREASE_ADDITIVE_TERM = 8.0; // Heuristic number of square meters to allow merging of smaller boxes
+
+    const union = box0.clone().union(box1);
```

Comment:
This causes unnecessary allocations. I suggest you pre-allocate once and reuse for every call. See https://github.com/cognitedata/reveal/blob/9e655cd98775e88e690b517e22f684a62eef91b0/viewer/packages/cad-geometry-loaders/src/sector/culling/OrderSectorsByVisibilityCoverage.ts#L276 for an example.